sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,867 @@
+[code]#host#, [code]#syclDevice# must either be contained by
+[code]#syclContext# or it must be a descendent device of some device that
+is contained by that context, otherwise this function throws a synchronous
+[code]#exception# with the [code]#errc::invalid# error code.
+
+Available only if `propertyListA` and `propertyListB` are specializations of the sycl::ext::oneapi::experimental::properties class.
+
+|====
+
+=== USM Memory Allocation Properties 
+
+This section specifies the properties that can be passed to the USM memory allocation functions defined in Section <<section.usm.allocs>>.
+A SYCL implementation or SYCL backend may support additional properties other than those defined here, provided they are defined in accordance with the
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties] extension.
+If any other properties are passed to the USM memory allocation functions with `properties` support, the compiler will issue a diagnostic error.
+
+All properties that are valid for annotated_ptr must be supported by the USM memory allocation functions with `properties` support.
+Unless otherwise specified, supported properties do not affect the behavior of the USM memory allocation functions with `properties` support.
+
+[NOTE]
+====
+The USM memory allocation functions defined in this extension support all properties supported by annotated_ptr, as a convenience. This allows the user to attach all desired properties to the annotated_ptr in a single step.
+These properties do not affect the behavior of the allocation functions, except where explicitly specified.
+====
+
+Table <<table.usm.malloc.properties>> lists properties that do affect the behavior of the allocation functions.
+
+[[table.usm.malloc.properties]]
+.Properties that affect the behavior of the USM memory allocation functions with properties support
+[options="header"]
+|====
+|Property|Description|Source
+a|
+alignment
+| 	
+Instructs the runtime to allocate memory starting at an address with this alignment in bytes.
+The address of the raw pointer belonging to the annotated_ptr returned by the USM memory allocation function will have this alignment in bytes. +
+|
+link:../proposed/sycl_ext_oneapi_annotated_ptr.asciidoc[sycl_ext_oneapi_annotated_ptr]
+|====
+
+Table <<table.usm.malloc.devhostsh>> lists the new properties introduced by this extension.
+The `usm_alloc` property is supported by annotated_ptr, and therefore by the USM memory allocation functions defined in this extension.
+This property will always appear on the annotated_ptr returned by the USM memory allocation functions defined in this extension that do not have a `usm::alloc` parameter.
+The `usm_alloc` property may also be passed to the USM memory allocation functions defined in this extension.
+If the USM memory allocation kind specified by a function's `usm::alloc` parameter is different from the `usm_alloc` property, the result is undefined behavior.
```

Comment:
This is easy to diagnose at compile-time, right?  If so, let's make it an "error, diagnostic required" instead.

Reply:
The usm::alloc parameter's value is not known until runtime. I will change this from UB to an exception.